crates/resolver/src/config.rs

```diff
@@ -819,6 +854,65 @@ impl<'a> ServerGroup<'a> {
     }
 }
 
+/// This is an alternate serialization function for a [`Duration`] that emits a single number,
+/// representing the number of seconds, instead of a struct with `secs` and `nanos` fields.
+#[cfg(feature = "serde")]
+fn duration_serialize<S>(duration: &Duration, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    duration.as_secs().serialize(serializer)
+}
+
+/// This is an alternate deserialization function for a [`Duration`] that expects a single number,
+/// representing the number of seconds, instead of a struct with `secs` and `nanos` fields.
+#[cfg(feature = "serde")]
+pub(crate) fn duration_deserialize<'de, D>(deserializer: D) -> Result<Duration, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Ok(Duration::from_secs(u32::deserialize(deserializer)?.into()))
+}
+
+/// This is an alternate serialization function for an optional [`Duration`] that emits a single
+/// number, representing the number of seconds, instead of a struct with `secs` and `nanos` fields.
+#[cfg(feature = "serde")]
+fn duration_opt_serialize<S>(duration: &Option<Duration>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    struct Wrapper<'a>(&'a Duration);
+
+    impl<'a> Serialize for Wrapper<'a> {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            duration_serialize(self.0, serializer)
+        }
+    }
+
+    match duration {
+        Some(duration) => serializer.serialize_some(&Wrapper(duration)),
+        None => serializer.serialize_none(),
+    }
```

Comment:
I think this could just call `duration_serialize(duration, serializer)` directly instead of bothering with the `Wrapper` setup?